### DISCUSSION TITLE
machines.override.json

### DISCUSSION THREAD

**Discussion by lfanton** [cockpit-project/cockpit#21120] [Q&A] [answered]
Hi community,

I am trying to modify the "machines.override.json" file to ensure that a specific user can only power on/off and access the VM console.
Is it possible to disable/hide unnecessary menus?

**Comment by martinpitt** [comment, marked as answer]
Please see https://cockpit-project.org/guide/latest/packages.html#package-manifest-override , that has an example how to hide menu itemss.

**Reply by lfanton** [reply]
Hi Martin,
While reading the guide, I created the various override files.
I tried using a file like this, but without success.

`{
  "menu": {
     "vms": {
        "label": "VM Manager",
        "add": false,
        "edit": false,
        "delete": false,
        "rename": false,
        "start": true,
        "stop": true,
        "virtual_console": true
     }
  }
}
`

**Reply by martinpitt** [reply]
what should all that "add"/"edit" etc. do? That looks like a derivation of cockpit-machines -- did you add code to it to look at these extra properties? The manifest files are only looked at by the Shell and cockpit-bridge, and they don't know about these extra fields. So it's not surprising that such an override makes absolutely no visible difference (with the standard cockpit-machines).

**Reply by jelly** [reply]
I think @lfanton wants to not allow a "guest login" to delete a virtual machine in the machines page. Cockpit does not support fine grained ACL's like this. And as a user can access the terminal page they can do whatever they want with `virsh` anyway.

**Reply by lfanton** [reply]
I've blocked access to Cockpit Terminal and other menus using override.json for all menus except 'Machines'.
I've also blocked SSH access.

My goal is to create a very simple interface for customers, allowing them to start/reboot the VM."

**Reply by lfanton** [reply]
@martinpitt You're right, that file is more like a concept map than a working example...

**Reply by martinpitt** [reply]
If you have control over the URL/domain (like, running cockpit behind a reverse proxy), there is also the possibility of just installing cockpit-machines and *not* cockpit-system (or other packages except cockpit-{bridge,ws}, and then only showing the machines page: http://localhost:9090/cockpit/@localhost/machines/index.html . See https://cockpit-project.org/guide/latest/embedding.html

**Reply by lfanton** [reply]
OK, so hiding the menu isn't enough...

**Comment by lfanton** [comment]
The documentation page is so poor!  Are there other sources?

**Reply by martinpitt** [reply]
No, that's it. There isn't really that much  else to say about them. It's a [JSON patch](https://jsonpatch.com/) for the package manifest, which is documented [here](https://cockpit-project.org/guide/latest/packages.html#package-manifest).
